src/libraries/ui/MigrateToGaslessModal.tsx

```diff
@@ -418,6 +169,274 @@ function MigrateToGasless() {
 			/>
 		</>
 	)
+
+	// useContexts defined here
+	const { waitForScwAddress, webwallet } = useContext(WebwalletContext)!
+	const { subgraphClients } = useContext(ApiClientsContext)!
+
+	// useStates start here
+	const [isOpen, setIsOpen] = useState(false)
+	const [isConnectWalletModalOpen, setIsConnectWalletModalOpen] = useState(false)
+	const [networkModalStep, setNetworkModalStep] = useState<number>()
+	const [transactionHash, setTransactionHash] = useState<string>()
+	const [shouldMigrate, setShouldMigrate] = useState<{state: number, chainId?: SupportedChainId}>()
```

Comment:
instead of state being a number, make it a literal type like `type MigrationState = 'success' | 'not-supported' ..`.  -- makes the code clearer to understand